types/cucumber/index.d.ts

```diff
@@ -56,7 +56,8 @@ export type AroundCode = (scenario: HookScenarioResult, runScenario?: (error: st
 export interface Transform {
     regexp: RegExp;
     transformer(arg: string): any;
-    typeName: string;
+    name?: string;
+    typeName?: string; // deprecated
```

Comment:
I would add a jsdoc to the deprecated property.

http://usejsdoc.org/tags-deprecated.html